### ISSUE TITLE
Topics occasionally remain unassigned. (Namespace is being unloaded, cannot add topic)

### ISSUE THREAD

**Issue by hozumit** [apache/pulsar#11920]
**Describe the bug**
I found that sometimes a topic remains unassigned after unloading.
The following message is repeatedly appeared in broker log.

```
08:28:28.538 [pulsar-io-4-15] WARN  org.apache.pulsar.broker.service.BrokerService - Namespace is being unloaded, cannot add topic persistent://mytenant/mynamespace/xxx-partition-37
```

The following log is the beginning of unloading.
```
09:57:13.634 [pulsar-web-40-7] INFO  org.apache.pulsar.broker.admin.impl.NamespacesBase - [null] Unloading namespace bundle mytenant/mynamespace/0x7ef23d0b_0x80000000
09:57:13.725 [pulsar-web-40-7] INFO  org.apache.pulsar.broker.namespace.OwnedBundle - Disabling ownership: mytenant/mynamespace/0x7ef23d0b_0x80000000
09:57:13.727 [pulsar-web-40-7] INFO  org.apache.pulsar.broker.service.BrokerService - [persistent://mytenant/mynamespace/xxx-partition-37] Unloading topic
```

```
$ docker exec -it pulsar_broker bin/pulsar-admin topics stats "persistent://mytenant/mynamespace/xxx-partition-37"
Can't find owner for topic persistent://mytenant/mynamespace/xxx-partition-37

Reason: Can't find owner for topic persistent://mytenant/mynamespace/xxx-partition-37
```

I saw this issue several times on different partitions of a topic, which have heavy loads.
I have solved the problem by restarting a broker process(docker container).

**To Reproduce**
Trigger unloading by command line( `bin/pulsar-admin namespaces unload mytenant/mynamespace` ) or broker load shedding.
Topic unloading and assignment mostly works. I think that it fails when brokers are in relatively heavy loads.

**Expected behavior**
I don't know if the allocation failure is a bug or not, but brokers should at least try to re-allocate the topic as soon as possible after the allocation failure.
I have solved the problem by restarting the broker process, so I think successive allocations will work.

**Desktop (please complete the following information):**
 - OS: CentOS
 - Pulsar (broker and client): v2.8.0

**Additional context**

**Comment by gaozhangmin** [comment]
which version you are using? this pr may fixed. #11310

**Comment by hozumit** [comment]
@gaozhangmin v2.8.0. Thank you for sharing the pr. It seems to be the same problem. I will wait v2.8.1 release.

**Comment by hozumit** [comment]
I've confirmed that the problem has been fixed in v2.8.1.
